### DISCUSSION TITLE
Widget contributed through Theia extension not showing

### DISCUSSION THREAD

**Discussion by LiamWalsh98** [eclipse-theia/theia#10882] [Q&A]
Hello,

I'm having a bit of trouble figuring out why my Theia plugin isn't being run properly when I click on the a file of the type I've associated with it. 

It seems that the plugin begins to launch, but then it arrives at this line in [@theia/core/src/browser/widget-manager.ts:222](https://github.com/eclipse-theia/theia/blob/66a7b8c4e8153f57ca8c8ac8d2754046c2fe4204/packages/core/src/browser/widget-manager.ts#L222). After which I can't seem to step into the code any further, and any subsequent breakpoints are never hit. The execution then finishes and nothing occurs in my Theia window. 

Is there anything I could do to help narrow down what the issue might be?

Thank you

**Comment by vince-fugnitto** [comment]
@LiamWalsh98 thank you for the discussion :+1: 

Do you happen to have additional information regarding the plugin? Perhaps it can be put in a public repository where others like myself can view the source-code and help you debug it?

If I understand correctly from your description, you are in fact creating a [theia plugin](https://theia-ide.org/docs/authoring_plugins/) (which has limited functionality) and not a [theia extension](https://theia-ide.org/docs/authoring_extensions/) (which can contribute additional functionality and allow you to rebind, and so on).

In offline discussions I believe you are basing your implementation off of https://github.com/eclipse-emfcloud/ecore-glsp which is not a theia plugin, but a theia extension as far as I am aware.

**Reply by LiamWalsh98** [reply]
I think you're right, it's an extension. not a plugin. Sorry, not totally comfortable with the terminology just yet. Thanks.

I was hesitant to push non-working code but I guess there's no harm; I'll put it up. Thank you for the response!

**Reply by LiamWalsh98** [reply]
Apologies for the rough state the repository is in but I just pushed my code to [rtpoet-dsl-diagram](https://github.com/LiamWalsh98/rtpoet-dsl-diagram). The Theia module can be found at [client/rtpoet-theia](https://github.com/LiamWalsh98/rtpoet-dsl-diagram/tree/master/client/rtpoet-theia). 

For what it's worth, the line in the glsp-theia-integration that my implementation never gets past is [glsp-diagram-manager:115](https://github.com/eclipse-glsp/glsp-theia-integration/blob/3655cc2dcb7836fcb191e96784bcea645b56664e/packages/theia-integration/src/browser/diagram/glsp-diagram-manager.ts#L115). And further into the callstack, this ([glsp-client-contribution:77](https://github.com/eclipse-glsp/glsp-theia-integration/blob/3655cc2dcb7836fcb191e96784bcea645b56664e/packages/theia-integration/src/browser/glsp-client-contribution.ts#L77)) is the last line i'm able to traverse before the extension launching process seems to end.

**Reply by vince-fugnitto** [reply]
@LiamWalsh98 do you have any additional information on how to build and start the repo with reproduction steps for your use-case? I'm wondering if the https://github.com/eclipse-emfcloud/emfcloud/discussions repo would be a better fit with specifics to `glsp` (it's not theia itself that maintains this repo).

As for debugging, given it is not actually a theia plugin as you've highlighted, debugging should be pretty standard. As for debugging promises you may want to look online for resources that may help.

**Reply by LiamWalsh98** [reply]
I believe it will build by running `./gradlew build` in the `server/` and then `yarn` in `client/` (which will also copy over the server jar). Then `yarn start:browser` should run the front end on localhost:3000 (Although I haven't tested a fresh build from a fresh clone. I can do so now). Sorry, I should have included this in a readme. 

Hmm, yes, maybe it's more appropriate to post the info there now that I have a more accurate picture of what exactly is failing. Nonetheless, I very much appreciate your help and responsiveness!

**Comment by paul-marechal** [comment]
The line that is blocking means something in your widget factory is not completing, you should look there.

https://github.com/eclipse-theia/theia/blob/66a7b8c4e8153f57ca8c8ac8d2754046c2fe4204/packages/core/src/browser/widget-manager.ts#L220-L222

**Reply by LiamWalsh98** [reply]
I wonder if there's a way for me to view whether or not this promise is actually being resolved. As of now I can only see the promises as "pending", so I can't tell if the problem lies there or elsewhere. 

Thank you for the info!
